bot/cogs/bot.py

```diff
@@ -100,81 +100,220 @@ def __init__(self, bot: AutoShardedBot):
 
     async def on_message(self, msg: Message):
-        if not msg.author.bot:
-            if msg.channel.id in self.channel_cooldowns:
-                on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
-                if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
-                    try:
-                        content = self.codeblock_stripping(msg.content)
-                        if not content:
+        if msg.channel.id in self.channel_cooldowns and not msg.author.bot:
+            on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
+            if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
+                try:
+                    not_backticks = ["'''", '"""', "´´´", "‘‘‘", "’’’", "′′′", "“““", "”””", "″″″", "〃〃〃"]
+                    bad_ticks = msg.content[:3] in not_backticks
+                    if bad_ticks:
+                        ticks = msg.content[:3]
+                        content, repl_code = self.codeblock_stripping(f"```{msg.content[3:-3]}```", True)
+                        if content is None:
                             return
 
+                        if len(content) == 2:
+                            content = content[1]
+                        else:
+                            content = content[0]
+
+                        space_left = 204
+                        if len(content) >= space_left:
+                            current_length = 0
+                            lines_walked = 0
+                            for line in content.splitlines(keepends=True):
+                                if current_length+len(line) > space_left or lines_walked == 10:
+                                    break
+                                current_length += len(line)
+                                lines_walked += 1
+                            content = content[:current_length]+"#..."
+
+                        howto = ("It looks like you are trying to paste code into this channel.\n\n"
+                                 "You seem to be using the wrong symbols to indicate where the codeblock should start. "
+                                 f"The correct symbols would be \`\`\`, not `{ticks}`.\n\n"
+                                 "**Here is an example of how it should look:**\n"
+                                 f"\`\`\`python\n{content}\n\`\`\`\n\n**This will result in the following:**\n"
+                                 f"```python\n{content}\n```")
+                    else:
+                        howto = ""
+                        content, repl_code = self.codeblock_stripping(msg.content, False)
+                        if content is None:
+                            return
                         # Attempts to parse the message into an AST node.
                         # Invalid Python code will raise a SyntaxError.
-                        tree = ast.parse(content)
+                        tree = ast.parse(content[0])
```

Comment:
huh? this doesn't seem to make sense anymore since you're unpacking the tuple on line 265.